llvm/lib/Target/RISCV/RISCVVLOptimizer.cpp

```diff
@@ -1235,16 +1236,21 @@ std::optional<MachineOperand> RISCVVLOptimizer::checkUsers(MachineInstr &MI) {
       return std::nullopt;
     }
 
-    // Tied operands might pass through.
-    if (UserOp.isTied()) {
-      LLVM_DEBUG(dbgs() << "    Abort because user used as tied operand\n");
-      return std::nullopt;
-    }
-
     auto VLOp = getMinimumVLForUser(UserOp);
     if (!VLOp)
       return std::nullopt;
 
+    // If the user is a passthru, we will need to preserve it if its tail is
+    // demanded.
+    if (UserOp.isTied()) {
```

Comment:
Is it possible for UserOp to be a non-passthru tied def? If so, do we need to check `RISCVII::isFirstDefTiedToFirstUse(UserOp.getParent().getDesc());`?

Reply:
Not that I'm aware of, but I'll add an assert that. i.e. that if an op is tied, it's the first operand and it's tied to the def.